src/main/cpp/src/json_parser.hpp

```diff
@@ -550,13 +565,13 @@ class json_parser {
     char* copy_destination,
     write_style w_style)
   {
-    return try_parse_quoted_string(
-      str_pos,
-      '\"',
-      to_match_str_pos,  // match str pos, nullptr means do not match
-      to_match_str_end,  // match str end
-      copy_destination,  // copy destination while parsing, nullptr means do not copy
-      w_style);
+    return try_parse_quoted_string(str_pos,
+                                   '\"',
```

Comment:
Similarly, do not pass a string literal. Pass either `std::string` or `string_scalar`.

Reply:
The `\"` char will be on GPU stack, IMO, it's safe to put it in the parameter.
I'll test this PR from end to end after this [PR](https://github.com/NVIDIA/spark-rapids-jni/pull/1862) is done